src/main/java/org/jpeek/graph/XmlGraph.java

```diff
@@ -64,19 +72,24 @@ public XmlGraph(final Skeleton skeleton) throws IOException {
     /**
      * Builds the graph from the skeleton.
      * @param skeleton XML representation on whiwh to build the graph
+     * @param pname Package of the class this graph is for
+     * @param cname Class in the skeleton this graph is for
      * @return List of nodes
-     * @throws IOException If fails
-     * @todo #445:30min This method works only for skeletons with a single
-     *  class because it assigns all methods in the skeleton to the first class
-     *  it finds. Instead, it should iterate through classes and then through
-     *  methods of each class.
      */
-    private static List<Node> build(final Skeleton skeleton) throws IOException {
+    private static List<Node> build(
+        final Skeleton skeleton, final String pname, final String cname
+    ) {
         final Map<XML, Node> byxml = new org.cactoos.map.Sticky<>(
             method -> method,
             method -> new Node.Simple(
                 new XmlMethodSignature(
-                    skeleton.xml().nodes("//class").get(0),
+                    skeleton.xml()
+                        .nodes(
+                            new Joined("", "//package[@id=", pname).toString()
```

Comment:
@vzurauskas I think using a `FormattedText` would not make this much more readable instead of `Joined` here and in other places.

Reply:
@victornoel Good idea.